ctapipe/io/hdf5eventsource.py

```diff
@@ -608,33 +633,13 @@ def _generator(self):
                     simulated = data.simulation.tel[tel_id]
 
                 if DataLevel.DL1_IMAGES in self.datalevels:
-                    if key not in image_readers:
```

Comment:
This is a big-ish change that's not directly related to the muon tool. Best to remove it from this PR and open another one if this is what we want. (I'd say yes...)